src/sidebar/helpers/mentions.ts

```diff
@@ -0,0 +1,114 @@
+import type { Mention } from '../../types/api';
+
+/**
+ * Wrap all occurrences of @mentions in provided text into the corresponding
+ * special tag, as long as they are surrounded by "empty" space (space, tab, new
+ * line, or beginning/end of the whole text).
+ *
+ * For example: `@someuser` with the `hypothes.is` authority would become
+ *  `<a data-hyp-mention data-userid="acct:someuser@hypothes.is">@someuser</a>`
+ */
+export function wrapMentions(text: string, authority: string): string {
+  return text.replace(
+    // Capture both the potential empty character before the mention (space, tab
+    // or new line), and the term following the `@` character.
+    // When we build the mention tag, we need to prepend that empty character to
+    // avoid altering the spacing and structure of the text.
+    /(^|\s)@(\w+)(?=\s|$)/g,
+    (match, precedingWhitespace, username) => {
+      const tag = document.createElement('a');
+
+      tag.setAttribute('data-hyp-mention', '');
+      tag.setAttribute('data-userid', `acct:${username}@${authority}`);
+      tag.textContent = `@${username}`;
+
+      return `${precedingWhitespace}${tag.outerHTML}`;
+    },
+  );
+}
+
+/**
+ * Replace all mentions wrapped in the special `<a data-hyp-mention />` tag with
+ * their plain-text representation.
+ */
+export function unwrapMentions(text: string) {
+  const tmp = document.createElement('div');
+  tmp.innerHTML = text;
+  for (const node of tmp.querySelectorAll('a[data-hyp-mention]')) {
+    node.replaceWith(node.textContent ?? '');
```

Comment:
The `textContent` for an HTML element is always a string, so you could use `node.textContent!` here. The reason it is nullable is because the property comes from `Node` and there are non-element subclasses for which `textContent` is null.